daft/dataframe/dataframe.py

```diff
@@ -479,6 +479,24 @@ def distinct(self) -> "DataFrame":
         builder = self._builder.distinct()
         return DataFrame(builder)
 
+    @DataframePublicAPI
+    def sample(self, fraction: float, with_replacement: bool = False, seed: Optional[int] = None) -> "DataFrame":
+        """Samples a fraction of rows from the DataFrame
+
+        Example:
+            >>> sampled_df = df.sample(0.5)
+
+        Args:
+            fraction (float): fraction of rows to sample.
+            with_replacement (bool, optional): whether to sample with replacement. Defaults to False.
+            seed (Optional[int], optional): random seed. Defaults to None.
+
+        Returns:
+            DataFrame: DataFrame with a fraction of rows.
+        """
+        builder = self._builder.sample(fraction, with_replacement, seed)
```

Comment:
We should ensure that `0.0 <= fraction <= 1.0` somewhere on the dataframe side as well. Right now I only see that check on the execution side.

Reply:
added this check in the public method for the Dataframe API as well